pkg/diagnostics/grpc_tracing.go

```diff
@@ -127,25 +93,15 @@ func FromGRPCContext(ctx context.Context) (trace.SpanContext, bool) {
 	return sc, ok
 }
 
-// AppendToOutgoingGRPCContext appends binary serialized SpanContext to the outgoing GRPC context
-func AppendToOutgoingGRPCContext(ctx context.Context, spanContext trace.SpanContext) context.Context {
+// SpanContextToGRPCMetadata appends binary serialized SpanContext to the outgoing GRPC context
+func SpanContextToGRPCMetadata(ctx context.Context, spanContext trace.SpanContext) context.Context {
+	if (spanContext == trace.SpanContext{}) {
```

Comment:
What are we testing here?

Reply:
trace is disabled and spanContext is empty. so no ops.

Reply:
You probably need parentheses here for the equality check:

https://stackoverflow.com/questions/28447297/how-to-check-for-an-empty-struct

Reply:
this is the way OC code uses. but I think I have the better way.

Reply:
I used `(trace.SpanContext{}) == spanContext` which is one of option in the link you refer to.

and added unit-tests for this.